bionic/task_state.py

```diff
@@ -74,10 +76,20 @@ def complete(self, task_key_logger):
         self.is_complete = True
 
     def get_results_assuming_complete(self, task_key_logger):
-        "Returns the results of an already-completed task state."
+        """
+        Returns the results of an already-completed task state. This can happen
+        by returning the cached results if they exist, or by computing the results
+        using `self._compute()` and returning them.
+        """
 
         assert self.is_complete
 
+        # If task state should persist but results aren't cached, that's probably
```

Comment:
Should this say "should *not* persist"? Also, "cached" is ambiguous here -- "memoized" may be clearer.

Reply:
Yeah, this should say should _not_ persist.